### ISSUE TITLE
support sending gzipped files?

### ISSUE THREAD

**Issue by arkarkark** [hugokernel/micropython-nanoweb#2]
I can send you a pull request because I already implemented this. Not sure if you'd be interested in it though.

I'm putting a react/tailwind app on my ESP01. Space is limited. I've found I can compress the built javascript file from 130k down to 42k with gzip. I modified nanoweb to first look for the requested file but with `.gz` appended then send the file with the Content-Encoding and Content-Type. Then all I need to do is upload app.js.gz rather than app.js and everything works.

Obviously this only works if you know all the browsers connecting to your device will `Accept-Encoding: gzip` I could check for it but I won't have the unzipped version anyway.

My changes add about 1k (41 lines) to the nanoweb file. Almost all of that is the annoying content-type code.

**Comment by hugokernel** [comment]
Hi arkarkark, any news about PR ? Thx

**Comment by arkarkark** [comment]
Sorry I was on vacation. I've been thinking a lot about how to make this more pluggable and also memory smart. I'd really like to not leave around the old send_file in memory if nothing is going to use it. I think what I'd like to do is make send_file and send_error etc be static members of the Nanoweb class that way when you override them the memory they used is freed up.

I also added a Cache-Control header which required mime types (as did my gzip option).

I also used a `readinto` to read into a buffer rather than constantly allocating memory blocks for sending large files.

I'll put together a pull request once we work out if I can keep [black turned on](https://github.com/hugokernel/micropython-nanoweb/pull/4) for code formatting.
